File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        #first convert all letters to lowercase and remove alphanumeric
        s = s.lower()

        #two pointers going from left and right side of the string to meet in middle
        r = 0
        l = len(s) - 1

        while r < l:
            if s[r].isalnum() == False or s[r] == ' ':
                r += 1
                continue
            if s[l].isalnum() == False or s[l] == ' ':
                l -= 1
                continue
            if s[r] == s[l]:
                r += 1
                l -= 1
                continue
            else:
                return False

        return True

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        // first convert all letters to lowercase
        std::transform(s.begin(),s.end(),s.begin(),::tolower);
        // two pointers, one from left the other from right
        int l = 0;
        int r = s.size() - 1;
        while(l < r){
            // check if its NOT alphanum
            while(l < r && !isalnum(s[l]))l++;
            while(l < r && !isalnum(s[r]))r--;
            // now compare characters
            if(s[l] != s[r]){
                return false;
            }
            l++;
            r--;
        }
        return true;
    }
};